packages/paste-core/components/timeline/src/types.ts

```diff
@@ -0,0 +1,99 @@
+import type { BoxProps } from "@twilio-paste/box";
+import type { IconWrapperProps } from "@twilio-paste/icons/esm/helpers/IconWrapper";
+import type { GenericIconProps } from "@twilio-paste/icons/esm/types";
+import type { HTMLPasteProps } from "@twilio-paste/types";
+
+export type TimelineProps = {
+  /**
+   * Overrides the default element name to apply unique styles with the Customization Provider.
+   *
+   * @default "TIMELINE"
+   * @type {BoxProps["element"]}
+   * @memberof TimelineProps
+   */
+  element?: BoxProps["element"];
+} & HTMLPasteProps<"ol">;
+
+export type TimelineItemProps = {
+  /**
+   * Title to display in the timeline item
+   *
+   * @type {string}
+   * @memberof TimelineItemProps
+   * @required
+   */
+  title: string;
+
+  /**
+   * Pass a Paste Icon to display in place of dot
+   *
+   * @type {React.FC<React.PropsWithChildren<GenericIconProps>>}
+   * @memberof TimelineItemProps
+   */
+  icon?: React.FC<React.PropsWithChildren<GenericIconProps>>;
+
+  /**
+   * Timestamp to display in the timeline item
+   *
+   * @type {string}
+   * @memberof TimelineItemProps
+   */
+  timestamp?: string;
+
+  /**
+   * If true, the timeline item will be collapsible
+   *
+   * @type {boolean}
+   * @memberof TimelineItemProps
+   * @default false
+   */
+  collapsible?: boolean;
+
+  /**
+   * In case of collapsible timeline item and no timestamp, this will be used as the heading
+   *
+   * @type {boolean}
+   * @memberof TimelineItemProps
+   * @default false
+   */
+  collapsibleHeading?: string;
+
+  /**
+   * Overrides the default element name to apply unique styles with the Customization Provider.
+   *
+   * @default "TIMELINE_ITEM"
+   * @type {BoxProps["element"]}
+   * @memberof TimelineItemProps
+   */
+  element?: BoxProps["element"];
+} & HTMLPasteProps<"li">;
+
+export type TimelineItemIconProps = {
```

Comment:
(suggestion): since we don't need to export TimelineItemIcon as it's not a part of the API, you can remove all the comments from this type. we only use them for creating the API page of the docs site, but this type won't show up there.